### ISSUE TITLE
Compilation error with nvcc 9.2.88 / gcc 7.2.0

### ISSUE THREAD

**Issue by bgranzow** [ibaned/omega_h#265]
Not really sure what this is about.
I was able to compile w/ gcc 7.3.0 on a CPU machine.
Maybe this is some nvcc nonsense?

```
[ 60%] Building CXX object src/CMakeFiles/unit_math.dir/unit_math.cpp.o
/ascldap/users/bngranz/ride/src/omega_h/src/Omega_h_few.hpp: In function ‘Omega_h::Matrix<m, m> Omega_h::log_glp(Omega_h::Matrix<m, m>) [with int dim = 1]’:
/ascldap/users/bngranz/ride/src/omega_h/src/Omega_h_few.hpp:35:39: error: array subscript is above array bounds [-Werror=array-bounds]
   OMEGA_H_INLINE T& operator[](Int i) { OMEGA_H_FEW_AT; }
                               ~~~~~~~~^
/ascldap/users/bngranz/ride/src/omega_h/src/Omega_h_few.hpp:35:39: error: array subscript is above array bounds [-Werror=array-bounds]
   OMEGA_H_INLINE T& operator[](Int i) { OMEGA_H_FEW_AT; }
                               ~~~~~~~~^
/ascldap/users/bngranz/ride/src/omega_h/src/Omega_h_few.hpp:35:39: error: array subscript is above array bounds [-Werror=array-bounds]
   OMEGA_H_INLINE T& operator[](Int i) { OMEGA_H_FEW_AT; }
                               ~~~~~~~~^
```

**Comment by bgranzow** [comment]
Apparently this compilation error goes away when setting:

```
-DOmega_h_CHECK_BOUNDS=ON
```

**Comment by ibaned** [comment]
I think I fixed this as of `v9.19.0`, there was a theoretical out-of-bounds for 1x1 matrix eigendecomposition (it would never have actually executed the out-of-bounds read, but it was too complex for the compiler to prove that this is the case).
